Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -0,0 +1,1232 @@
+              }
+
+    ruleset = client.rule_sets.get_by_reference(ruleset_href)
+    ruleset_json = ruleset.to_json()
+    for rules in ruleset_json.get("rules", []):
+        existing_rule = {"ingress_services": sorted([href.get('href') for href in rules.get("ingress_services", {})]),
+                         "providers": sorted(extract_values_from_dictionary(rules.get("providers"))),
+                         "consumers": sorted(extract_values_from_dictionary(rules.get("consumers"))),
+                         "resolve_providers_as": sorted(rules.get("resolve_labels_as").get("providers")),
+                         "resolve_consumers_as": sorted(rules.get("resolve_labels_as").get("consumers"))
+                         }
+        if params == existing_rule:
+            demisto.info("Found existing Rule bounded to the Ruleset: {}.".format(ruleset_href))
+            rule_href = rules.get("href")
+            response = client.rules.get_by_reference(rule_href)
+            break
+    else:
+        rule = Rule.build(
+            ingress_services=ingress_services,
+            consumers=consumers, providers=providers,
+            resolve_consumers_as=resolve_consumers_as,
+            resolve_providers_as=resolve_providers_as
+        )
+        response = client.rules.create(rule, parent=ruleset_href)
+
+    response = response.to_json()
+    readable_output = prepare_rule_create_output(response)  # type: ignore
+
+    return CommandResults(
+        outputs_prefix="Illumio.Rule",
+        outputs_key_field="href",
+        outputs=remove_empty_elements(response),
+        readable_output=readable_output,
+        raw_response=response
+    )
+
+
+def main():
+    """Parse params and runs command functions."""
+    try:
+        command = demisto.command()
+
+        params = demisto.params()
+        api_user = params.get("api_user")
+        api_key = params.get("api_key")
+
+        port = arg_to_number(params.get("port"), required=True, arg_name="port")
+        if port < MIN_PORT or port > MAX_PORT:  # type: ignore
+            raise InvalidValueError(
+                message="{} is an invalid value for port. Value must be in 1 to 65535.".format(port))
+
+        org_id = arg_to_number(params.get("org_id"), required=True, arg_name="org_id")
+        if org_id <= 0:  # type: ignore
+            raise ValueError(
+                "{} is an invalid value. Organization ID must be a non-zero and positive numeric value.".format(
+                    org_id
+                )
+            )
+
+        base_url = params.get("url").strip()
+        if not base_url:
+            raise ValueError("Server URL is required.")
+
+        proxy = handle_proxy()
+
+        client = PolicyComputeEngine(url=base_url, port=port, org_id=org_id)
+        client.set_proxies(
+            http_proxy=proxy.get("http", None), https_proxy=proxy.get("https", None)
```

Comment:
- if None is your default value then you don’t need to put it here (`{'a': 'b'}.get('c')` is equal to `{'a': 'b'}.get('c', None)`.
- more important is that in case the proxy is not settled, `handle_proxy()` will return `{'http': '', 'https': ''}` so you need to check it because it might lead to problems in case the SDK checks it explicitly against `None`.